src/main/java/fr/zcraft/quartzlib/components/gui/PromptGui.java

```diff
@@ -56,39 +57,52 @@
 
     /* ===== Reflection to Sign API ===== */
     private static Field fieldTileEntitySign = null; // 1.11.2-: CraftSign.sign; 1.12+: CraftBlockEntityState.tileEntity
+    private static Object tileEntitySign = null; //1.17+
     private static Field fieldTileEntitySignEditable = null; // 1.12+ only: CraftBlockEntityState.isEditable
     private static Method methodGetHandle = null; // CraftPlayer.getHandle()
     private static Method methodOpenSign = null; // EntityHuman.openSign()
     private final Callback<String> callback;
     private Location signLocation;
     private String contents;
 
+
     public PromptGui(Callback<String> callback, String contents) {
         this(callback);
         this.contents = contents;
     }
 
     /**
      * Creates a new prompt GUI, using the given callback.
+     *
      * @param callback The callback to be given the input text to.
      */
     public PromptGui(Callback<String> callback) {
         super();
+
         if (!isAvailable()) {
             throw new IllegalStateException("Sign-based prompt GUI are not available");
         }
 
         this.callback = callback;
     }
 
+
     /**
      * Checks if Prompt GUIs can be correctly used on this Minecraft versions.
      */
     public static boolean isAvailable() {
-        if (!isInitialized) {
-            init();
+        try {
+            if (!isInitialized) {
+                init();
+            }
+            return fieldTileEntitySign != null;
+        } catch (Exception e) {
+            //1.17+ can't use the fields because of java 17
+            if (!isInitialized) {
+                init();
+            }
+            return tileEntitySign != null;
```

Comment:
Cannot this part be extracted out of the try/catch?

```java
if (!isInitialized) {
    init();
}
```

Reply:
I simply removed the try catch and the second part that was useless this only is usefull with very old version of minecraft (1.11-)